### PR TITLE
WRO-735: Add guide document for ui and screenshot tests when contributing

### PR DESCRIPTION
**Pull request by dongsuwon**
### Checklist

* [X] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A [CHANGELOG entry](http://enactjs.com/docs/developer-guide/contributing/changelogs/) is included
* [ ] At least one test case is included for this feature or bug fix
* [X] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Added guide documentation for UI and Screenshot tests when contributing

### Links
[//]: # (Related issues, references)
WRO-735

Enact-DCO-1.0-Signed-off-by: Dongsu Won (dongsu.won@lgepartner.com)

### CODE REVIEW

File: docs/testing-components/index.md
Comment (seunghoh) [inline docs/testing-components/index.md:8]: Ui Testing -> UI Testing

File: docs/testing-components/ui-testing/index.md
Comment (seunghoh) [inline docs/testing-components/ui-testing/index.md:2]: Ui Testing -> UI Testing

File: docs/testing-components/ui-testing/index.md
Comment (seunghoh) [inline docs/testing-components/ui-testing/index.md:11]: Ui Testing -> UI Testing

File: docs/testing-components/screenshot-testing/index.md
Comment (seunghoh) [inline docs/testing-components/screenshot-testing/index.md:8]: tests or Tests?

File: docs/testing-components/screenshot-testing/index.md
Comment (seunghoh) [inline docs/testing-components/screenshot-testing/index.md:5]: Let's add What is the Screenshot Test?

File: docs/testing-components/ui-testing/index.md
Comment (seunghoh) [reply docs/testing-components/ui-testing/index.md:90]: Let's move the top of the list and use title "What is the UI Testing"

**Review by seunghoh — CHANGES_REQUESTED** [review]

File: docs/testing-components/index.md
Comment (dongsuwon) [reply docs/testing-components/index.md]: Suggested changes.

File: docs/testing-components/ui-testing/index.md
Comment (dongsuwon) [reply docs/testing-components/ui-testing/index.md]: Suggested changes.

File: docs/testing-components/ui-testing/index.md
Comment (dongsuwon) [reply docs/testing-components/ui-testing/index.md]: It is move to 'What is the UI Testing'.

File: docs/testing-components/index.md
Comment (dongsuwon) [reply docs/testing-components/index.md]: Suggested change.

File: docs/testing-components/screenshot-testing/index.md
Comment (dongsuwon) [reply docs/testing-components/screenshot-testing/index.md]: Suggested change.

File: docs/testing-components/screenshot-testing/index.md
Comment (dongsuwon) [reply docs/testing-components/screenshot-testing/index.md:4]: Suggested change.

File: docs/testing-components/ui-testing/index.md
Comment (dongsuwon) [reply docs/testing-components/ui-testing/index.md]: Suggested change.

File: docs/testing-components/screenshot-testing/index.md
Comment (jeonghee27) [inline docs/testing-components/screenshot-testing/index.md:6]:
```suggestion
Screenshot test is visual test that takes a screenshot, then compares it to a reference screenshot file saved. When ui changes, one of the major considerations is whether visual changes exist. Some changes may be easy to check, but some changes are hard to check due to very small change. With Screenshot test, very small changes can easily check due to compare pixel-by-pixel. Additionally, Screenshot testing is the most reliable test for static images. If you want to test animated image or video, you should check the result.
```

I don't understand your intend for the last sentence

I don't know your intend for last sentence
"If you want to test animated image or video, you should check result."

File: docs/testing-components/screenshot-testing/index.md
Comment (jeonghee27) [inline docs/testing-components/screenshot-testing/index.md:12]:
```suggestion
1. Add @enact/ui-test-utils as a devDependency: `npm i --save-dev @enact/ui-test-utils`.
```

File: docs/testing-components/screenshot-testing/index.md
Comment (jeonghee27) [inline docs/testing-components/screenshot-testing/index.md:14]: Sorry I cannot understand the mean of "within the UI libraray"

File: docs/testing-components/ui-testing/index.md
Comment (jeonghee27) [inline docs/testing-components/ui-testing/index.md:41]: I think it's better to include this by default in CLI. We need discuss

File: docs/testing-components/ui-testing/index.md
Comment (jeonghee27) [inline docs/testing-components/ui-testing/index.md:53]: Without examples of View files and Spec files, App developers wouldn't be able to create them.

File: docs/testing-components/index.md
Comment (jeonghee27) [inline docs/testing-components/index.md:7]: The original ticket intent was to add these to the `contributing` page.
I think we should provide more default setting to run tests in enact apps. 
I willing to do this, but we need the opinion of others.

File: docs/contributing/index.md
Comment (jeonghee27) [inline docs/contributing/index.md:61]: *   Ensure all new features have unit tests, [ui tests](./ui-testing.md), and /[screenshot tests](./screenshot-testing.md) that pass and any bug fixes include tests that fail without the fix applied. 

I think we should add ui/screesnhot test next to unit test.

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:21]: enact link is optional step to develop enact other lib (enact/core, enact/ui, ui-test-utils...). Let's remove it.

```suggestion
```

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:43]: the result was broken. please check nebular

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:86]:
```suggestion
Screenshot testing is a test that compares and assures two different outputs. For the first test without any reference, all results are saved in the `tests/screenshot/dist/reference` and they become reference for later tests. In the second run, all results are saved in the `tests/screenshot/dist/screen` and compare screen with reference.
```

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:64]: Is there any place to put a github link? (see detail? )

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:62]:
```suggestion
The `generateTestData` and `Page` component from `@enact/ui-test-utils/utils/` generateTestData and Page contain useful methods for screenshot tests.
```
But I'm not sure if this part is really necessary.

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:71]:
```suggestion
* Re-run tests without build:
```

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:75]:
```suggestion
* Running with filtering by component and without build:
```

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:72]:
```suggestion
The Page component from `@enact/ui-test-utils/utils/` Page contains useful methods for writing tests.
```

I don't know what is mean of loading.

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:96]:
```suggestion
Also, you can see the view you made on chrome. This requires that a server be running on port 5000. If you have globally installed the serve command with `npm install -g serve` you can start the server like this `serve ./test/ui/dist/`.
```

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:97]:
```suggestion
To open a specific test app, open the URL path for the test. A specific app is displayed as a directory on this page. You can see the app as click the app you wanted.
```

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:99]: It would be better if it is at the top like a screenshot test.

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:1]: I hope that the link below will also be included in somewhere of this document.
https://github.com/enactjs/ui-test-utils

File: docs/contributing/screenshot-testing.md
Comment (dongsuwon) [reply docs/contributing/screenshot-testing.md]: I removed this part and added link for ui-test-utils.

File: docs/contributing/ui-testing.md
Comment (dongsuwon) [reply docs/contributing/ui-testing.md]: I removed this part and added link for ui-test-utils.

File: docs/contributing/screenshot-testing.md
Comment (jeonghee27) [inline docs/contributing/screenshot-testing.md:61]: "GitHub methods" is little bit weird.

```suggestion
For details on the relevant APIs or options, please see [ui-test-utils](https://github.com/enactjs/ui-test-utils).
```

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:77]: Same with ss test

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:7]:
```suggestion
* How component handles user actions performed via input devices.
* Whether elements correctly work as intended.
```

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [inline docs/contributing/ui-testing.md:22]:
```suggestion
```

File: docs/contributing/screenshot-testing.md
Comment (dongsuwon) [reply docs/contributing/screenshot-testing.md]: Suggested Change.

File: docs/contributing/ui-testing.md
Comment (dongsuwon) [reply docs/contributing/ui-testing.md]: Suggested Change.

File: docs/contributing/ui-testing.md
Comment (dongsuwon) [reply docs/contributing/ui-testing.md]: Suggested Change.

File: docs/contributing/ui-testing.md
Comment (jeonghee27) [reply docs/contributing/ui-testing.md]: Please recheck it

Comment (dongsuwon) [reply docs/contributing/ui-testing.md]: Sorry, I missed it. Thank you!

**Review by jeonghee27 — APPROVED** [review]
LGTM

File: docs/contributing/index.md
Comment (jeonghee27) [inline docs/contributing/index.md:60]:
```suggestion
	*   Ensure all new features have unit tests,[ui test](./ui-testing.md) and [screenshot test](./screenshot-testing.md) that pass and any bug fixes include tests that fail without the fix applied.
```

Comment (seunghoh) [reply docs/contributing/index.md:60]: @jeonghee27 Do you have any specific reason for using terms "UI-Test" and "Screenshot"?

File: docs/contributing/ui-testing.md
Comment (seunghoh) [inline docs/contributing/ui-testing.md:2]: UI Testing

File: docs/contributing/ui-testing.md
Comment (seunghoh) [inline docs/contributing/ui-testing.md:4]: UI Testing. Let's be consistent with Unita Testing and Screenshot Testing

**Review by seunghoh — CHANGES_REQUESTED** [review]

File: docs/contributing/ui-testing.md
Comment (seunghoh) [inline docs/contributing/ui-testing.md:23]: Creating Tests

File: docs/contributing/ui-testing.md
Comment (seunghoh) [inline docs/contributing/ui-testing.md:5]: Enact supports ui-test-utils package which includes the common WebDriver configurations and some utility modules for executing automated UI tests. With UI test, you can execute user interaction testing quickly and reliably.

File: docs/contributing/ui-testing.md
Comment (seunghoh) [inline docs/contributing/ui-testing.md:7]: Not sure

**Review by seunghoh — CHANGES_REQUESTED** [review]

File: docs/contributing/ui-testing.md
Comment (seunghoh) [inline docs/contributing/ui-testing.md:35]: confused to match with the example above

**Review by seunghoh — CHANGES_REQUESTED** [review]

**Review by seunghoh — APPROVED** [review]

**Comment by seunghoh** [comment]
For the integrator's information.
- Before merging, the example code should be updated as I commented. Please make sure to update
- I changed some of the contents of the doc and found that It happened to be very similar to the readme in the ui-test-utils https://github.com/enactjs/ui-test-utils/blob/master/README.md
- So, I feel like not adding screenshot-testing, ui-testing md file and just add the link in the index.md like `Ensure all new features have unit tests that pass and any bug fixes include tests that fail without the fix applied. Adding automated [UI tests](https://github.com/enactjs/ui-test-utils/blob/master/README.md) and [Screenshot tests](https://github.com/enactjs/ui-test-utils/blob/master/README.md) supported by Enact packages are highly recommended.`
- But the final decision is on the integrator :)

File: docs/contributing/adding-automated-ui-tests.md
Comment (seunghoh) [inline docs/contributing/adding-automated-ui-tests.md:51]: Let's link more specific :)
https://github.com/enactjs/ui-test-utils/blob/master/README.md#filtering-tests

File: docs/contributing/adding-automated-ui-tests.md
Comment (seunghoh) [inline docs/contributing/adding-automated-ui-tests.md:60]: remove tab

File: docs/contributing/adding-automated-ui-tests.md
Comment (seunghoh) [inline docs/contributing/adding-automated-ui-tests.md:151]: https://github.com/enactjs/ui-test-utils/blob/master/README.md#filtering-screenshot-by-component

**Review by seunghoh — APPROVED** [review]

File: docs/contributing/adding-automated-ui-tests.md
Comment (MikyungKim) [inline docs/contributing/adding-automated-ui-tests.md:202]:
```suggestion
And add your component to the `importer.js` and `SandstoneComponents.js` files like:
```

Comment (dongsuwon) [reply docs/contributing/adding-automated-ui-tests.md]: Suggested change.

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**
